external/gtest/CMakeLists.txt

```diff
@@ -1,30 +1,43 @@
-configure_file(CMakeLists.txt.in CMakeLists.txt @ONLY)
+cmake_minimum_required(VERSION 3.10)
+project(GoogleTest CXX)
 
-configure_file(CMakeLists.txt.in CMakeLists.txt)
-execute_process(COMMAND "${CMAKE_COMMAND}" -G "${CMAKE_GENERATOR}" .
-  RESULT_VARIABLE result
-  WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
-)
-if(result)
-  message(FATAL_ERROR "CMake step for googletest failed: ${result}")
+if(WIN32)
+    set(gtest_cxx_flags -D_UNICODE -DUNICODE -DSTRICT -DWIN32_LEAN_AND_MEAN)
```

Comment:
why?

Reply:
Why specifically this flags for Windows?

Reply:
Why there is a need of some flags at all?

Reply:
LEAN_AND_MEAN - to fix build on new MSVC compilers (and generally increase build speed)
UNICODE - to make gtest use unicode-aware WinAPI
It also fixes some build warning

Reply:
Have you met these warnings on your machine?

Reply:
Yes, I wouldn't add them if I wouldn't

Reply:
BTW:
I didn't come-up with this flags just yesterday.
I've been using them in many other project for quite a while.